src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
+    private String time;
+    private LocalDate dateTime;
+    private boolean isParsedDate;
+
+    public Deadline(String line) throws EmptyStringException{
+        super();
+        if(line.isBlank()){
+            throw new EmptyStringException("Deadline cannot be empty.");
+        }
+        String[] command = line.split(" \\/by ");
+        this.item = command[0];
+        this.time = command[1];
+        this.isParsedDate = false;
+        try {
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+            this.dateTime = LocalDate.parse(this.time, format);
+            this.isParsedDate = true;
+        }
+        catch (Exception e){
+            //Date failed to parse
+        }
+        taskType = "D";
+    }
+
```

Comment:
Perhaps you could rename the encode() method in order to explain it's purpose? For example, maybe encodeDateFormat() would be a more descriptive way? Just a suggestion.